local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -111,6 +111,13 @@ func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConf
 		return
 	}
 
+	// Cache mount points to iterate through discovered paths (aka files) below
+	mountPoints, mountPointsErr := d.RuntimeConfig.Mounter.List()
```

Comment:
Did you check that this also returns bind mounts?

Reply:
After this change goes in, I'll need to tag a new version of the provisioner, and then the local e2e tests need to be updated to do bind mounts for directories.

Reply:
@msau42 - I ran the following test to verify that bind mounts are also returned inside the provisioner container.
```
1. mkdir /tmp/var2
2. mkdir -p /var/log/var2
3. mount --bind /tmp/var2 /var/log/var2
```
Next, I executed Mounter.List() and printed the following tuple for each mount point: `<mountPoint.Device:mountPoint.Path:mountPoint.Type>`

The discovered mount point list that was enumerated showed the following tuple:
`overlay:/var/log/var2:overlay`

Let me know if you have any questions.

Reply:
@msau42 - I'll open an issue and start working on this change to the e2e tests.